File: Electronics_Shop_17.Model/Requests/AddCustomerCoupon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Electronics_Shop_17.Model.Requests
{
    public class AddCustomerCoupon
    {
        public int CustomerId { get; set; }
        public int CouponId { get; set; }
        public int UsageCount { get; set; } = 0;        
    }
}

File: angular17_web/src/app/models/customer-coupon/add.customer.coupon.ts
export class AddCustomerCoupon {
  customerId!: number;
  couponId!: number;
  usageCount?: number;

  constructor(data?: any) {
    if (!data) return;

    this.customerId = data.customerId;
    this.couponId = data.couponId;
    this.usageCount = data.usageCount;
  }
}
